src/app/services/weather.service.ts

```diff
@@ -87,7 +89,15 @@ export class WeatherService {
 
   public readonly forecast$ = this._forecast$.asObservable();
 
-  constructor(private readonly httpClient: HttpClient) {
+  private cached = {
+    [StorageKeysEnum.CURRENT_WEATHER]: true,
+    [StorageKeysEnum.FORECAST_WEATHER]: true,
+  };
```

Comment:
Just for the record, I hate this, but I don't see a better way of approaching this.

If we were to initialize the behavior subjects with initial data from LS, we'd still get the loader, since the loading state is set to loading as soon as the action for fetching data from api or mocks is dispatched.

I could use skip() operator or some other rxjs black magic, but there's really no elegant solution, since the data gets fetched as soon as the page loads :/